src/components/Dashboard/Role/AddUser.js

```diff
@@ -0,0 +1,141 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { Button, Form } from 'react-bootstrap';
+import {
+  DashboardLink as DashLink,
+  DashboardLayout as Layout
+} from 'components/Dashboard/';
+import { UserRoles } from 'components/Dashboard/';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { actions as rolesActions } from 'reducers/roles';
+import { actions as dashActions } from 'reducers/dashboard';
+import { actions as usersActions } from 'reducers/users';
+import { getAllUsers } from 'selectors/users';
+
+export class RoleAddUser extends Component {
+  static propTypes = {
+    actions: PropTypes.object.isRequired,
+    name: PropTypes.string,
+    layoutOptions: PropTypes.object.isRequired,
+    roleId: PropTypes.number,
+    users: PropTypes.array
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = { userId: '0', submitting: false };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const { actions } = this.props;
+
+    actions.requestUsers();
```

Comment:
This is an instance where we'd ideally have a boolean `loaded` prop in the redux state tree that is set to true on `_SUCCESS` - to cut down on network calls if the user is moving between pages.